runtime/protocol/http-test/common/src/aws/smithy/kotlin/runtime/httptest/HttpTrafficParser.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.httptest
+
+import aws.smithy.kotlin.runtime.http.*
+import aws.smithy.kotlin.runtime.http.content.ByteArrayContent
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+import aws.smithy.kotlin.runtime.http.request.HttpRequestBuilder
+import aws.smithy.kotlin.runtime.http.request.url
+import aws.smithy.kotlin.runtime.http.response.HttpResponse
+import aws.smithy.kotlin.runtime.util.decodeBase64Bytes
+import kotlinx.serialization.json.*
+
+internal fun parseHttpTraffic(json: String) = buildTestConnection {
+    val traffic = Json.parseToJsonElement(json).jsonArray
```

Comment:
nit/suggestion

assuming empty string fails to parse maybe check that the input param is not empty for a better error